### ISSUE TITLE
Hang when latency peer can't speak to server

### ISSUE THREAD

**Issue by richb-hanover** [Zoxc/crusader#73]
I screwed up and reversed the addresses of the server and latency peer. I had intended to test my server on the internet, and use the latency peer (on the local subnet) in the test.

The result was a hang with no output after the connection messages. I had to abort the test.

```
[2024-09-11 07:01:54] Client version 0.2.1-dev (commit 40b3c7fed7a9bc2df06e294efe17e2dcb9dddf34) running
[2024-09-11 07:01:54] Connected to server 192.168.9.1:35481
[2024-09-11 07:01:54] Connected to peer 1.2.3.4:35481
[2024-09-11 07:02:33] Aborted...
```

I wonder if the latency peer can give some indication to the client that it cannot hear the server...

**Comment by Zoxc** [comment]
I added a shorter connection timeout, which would be a partial fix.

**Comment by richb-hanover** [comment]
I just tried and it seems to time out in 8-9 seconds. This is much better. Thanks

It would be great to give a better message than "Expected protocol message, but stream closed" in this instance. It doesn't give the viewer a clue about what to do to fix it. 

Perhaps "Latency peer couldn't communicate with server. Check their addresses."

**GUI Client**
```
[2024-09-13 07:43:04] Client version 0.2.1-dev (commit e9e5c3e289fe2ebcb866e3becb8fc4a9a3938dd3) running
[2024-09-13 07:43:04] Connected to server 192.168.253.158:35481
[2024-09-13 07:43:04] Connected to peer 5.78.66.2:35481
[2024-09-13 07:43:13] Error: Expected protocol message, but stream closed
```

**Latency peer**

```
[2024-09-13 07:42:37] Server version 0.2.1-dev (commit e9e5c3e289fe2ebcb866e3becb8fc4a9a3938dd3) running...
[2024-09-13 07:42:37] Address on `eth0`: 2a01:4ff:1f0:e512::1
[2024-09-13 07:42:37] Address on `eth0`: 5.78.66.2

[2024-09-13 07:42:51] Serving as peer for 198.55.239.72:64924, version 3
[2024-09-13 07:42:51] Server for peer is 192.168.253.158:35481
[2024-09-13 07:42:59] Error serving client 198.55.239.72:64924: Failed to run peer

Caused by:
    Timed out trying to connect to server. Is the server running?
[2024-09-13 07:43:05] Serving as peer for 198.55.239.72:64927, version 3
[2024-09-13 07:43:05] Server for peer is 192.168.253.158:35481
[2024-09-13 07:43:13] Error serving client 198.55.239.72:64927: Failed to run peer

Caused by:
    Timed out trying to connect to server. Is the server running?
```

**Comment by richb-hanover** [comment]
This seems to have been addressed in `0.3`
